drivers/pinctrl/Kconfig.esp32

```diff
@@ -0,0 +1,9 @@
+# Copyright (c) 2022 Espressif Systems (Shanghai) Co., Ltd.
+# SPDX-License-Identifier: Apache-2.0
+
+config PINCTRL_ESP32
+	bool "Pin controller driver for ESP32 SoC"
+	depends on SOC_ESP32
+	default y
```

Comment:
please default based on `dt_compat_status`

Reply:
Done.

Reply:
can't see this fixed in the first commit, can you fixup this one?

Reply:
Fixed.